scripts/generate-google3-docs/resource-reference/main.go

```diff
@@ -852,7 +852,22 @@ func crdFilePathForGVK(gvk schema.GroupVersionKind) string {
 }
 
 func crdFileNameForGVK(gvk schema.GroupVersionKind) string {
-	crdName := fmt.Sprintf("%s.%s", text.Pluralize(gvk.Kind), gvk.Group)
+	plural := text.Pluralize(gvk.Kind)
+
+	// special plural formats. The best practice is to use singular as is
+	alreadyPluralWords := []string{"settings", "metrics", "series", "data"}
```

Comment:
This logic is the same as API validation on singular/plural format https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/tests/apichecks/crds_test.go#L810

Reply:
I previously added a special case for "IAPSettings" in `text.Pluralize()`. maybe we should consolidate the special case handling?